boards/feather-m0/include/arduino_pinmap.h

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (C) 2021 J. David Ibáñez <jdavid.ibp@gmail.com>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     boards_feather-m0
+ * @{
+ *
+ * @file
+ * @brief       Mapping from MCU pins to Arduino pins
+ *
+ * You can use the defines in this file for simplified interaction with the
+ * Arduino specific pin numbers.
+ *
+ * @author      J. David Ibáñez <jdavid.ibp@gmail.com>
+ */
+
+#ifndef ARDUINO_PINMAP_H
+#define ARDUINO_PINMAP_H
+
+#include "periph/gpio.h"
+#include "periph/adc.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @name    Mapping of MCU pins to Arduino pins
+ *
+ * @{
+ */
+
+#define ARDUINO_PIN_0           GPIO_PIN(PA, 11) /**< D0 (RX) */
+#define ARDUINO_PIN_1           GPIO_PIN(PA, 10) /**< D1 (TX) */
+#define ARDUINO_PIN_2           GPIO_UNDEF       /**< D2 */
+#define ARDUINO_PIN_3           GPIO_UNDEF       /**< D3 */
+#define ARDUINO_PIN_4           GPIO_UNDEF       /**< D4 */
+#define ARDUINO_PIN_5           GPIO_PIN(PA, 15) /**< D5 */
+#define ARDUINO_PIN_6           GPIO_PIN(PA, 20) /**< D6 */
+#define ARDUINO_PIN_7           GPIO_UNDEF       /**< D7 */
+#define ARDUINO_PIN_8           GPIO_UNDEF       /**< D8 */
+#define ARDUINO_PIN_9           GPIO_PIN(PA, 7)  /**< D9 */
+#define ARDUINO_PIN_10          GPIO_PIN(PA, 18) /**< D10 */
+#define ARDUINO_PIN_11          GPIO_PIN(PA, 16) /**< D11 */
+#define ARDUINO_PIN_12          GPIO_PIN(PA, 19) /**< D12 */
+#define ARDUINO_PIN_13          GPIO_PIN(PA, 17) /**< D13 (LED) */
+#define ARDUINO_PIN_14          GPIO_PIN(PA, 2)  /**< D14 (A0) */
+#define ARDUINO_PIN_15          GPIO_PIN(PB, 8)  /**< D15 (A1) */
+#define ARDUINO_PIN_16          GPIO_PIN(PB, 9)  /**< D16 (A2) */
+#define ARDUINO_PIN_17          GPIO_PIN(PA, 4)  /**< D17 (A3) */
+#define ARDUINO_PIN_18          GPIO_PIN(PA, 5)  /**< D18 (A4) */
+#define ARDUINO_PIN_19          GPIO_PIN(PB, 2)  /**< D19 (A5) */
+#define ARDUINO_PIN_20          GPIO_PIN(PA, 22) /**< D20 (I2C SDA) */
+#define ARDUINO_PIN_21          GPIO_PIN(PA, 23) /**< D21 (I2C SCL) */
+#define ARDUINO_PIN_22          GPIO_PIN(PA, 12) /**< D22 (SPI MISO) */
+#define ARDUINO_PIN_23          GPIO_PIN(PB, 10) /**< D23 (SPI MOSI) */
+#define ARDUINO_PIN_24          GPIO_PIN(PB, 11) /**< D24 (SPI SCK) */
```

Comment:
As said, on WiFI and LoRa variants of this board, the Arduino pins D2, D3, D4, D7, and D8 are used to control additional hardware modules. Although these pins are not broken out on these boards, it might be useful to have them defined for compatibility with Arduino libraries.

Therefore, I'm wondering whether it might make sense to change it as follows. Otherwise, all pins would have to be defined again and again for each boards which in fact use `feather-m0` as base board.

```suggestion
#if defined(BOARD_FEATHER_M0_WIFI)
#define ARDUINO_PIN_2           GPIO_PIN(PA, 14) /**< D2 (WINC_CHIP_EN) */
#define ARDUINO_PIN_3           GPIO_UNDEF       /**< D3 */
#define ARDUINO_PIN_4           GPIO_PIN(PA, 8)  /**< D4 (WINC_RST */
#define ARDUINO_PIN_7           GPIO_PIN(PA, 21) /**< D7 (WINC_IRQ */
#define ARDUINO_PIN_8           GPIO_PIN(PA, 6)  /**< D8 (WINC_CS */
#elif defined(BOARD_FEATHER_M0_WIFI)
#define ARDUINO_PIN_2           GPIO_UNDEF       /**< D2 */
#define ARDUINO_PIN_3           GPIO_PIN(PA, 9)  /**< D3 (LORA_IRQ)
#define ARDUINO_PIN_4           GPIO_PIN(PA, 8)  /**< D4 (LORA_RST
#define ARDUINO_PIN_7           GPIO_UNDEF       /**< D7 */
#define ARDUINO_PIN_8           GPIO_PIN(PA, 6)  /**< D8 (LORA_CS) */
#elif defined(BOARD_FEATHER_M0_BLE)
#define ARDUINO_PIN_2           GPIO_PIN(PA, 14) /**< D2 (WINC_CHIP_EN) */
#define ARDUINO_PIN_3           GPIO_UNDEF       /**< D3 */
#define ARDUINO_PIN_4           GPIO_PIN(PA, 8)  /**< D4 (BLE_SWDIO/RST) */
#define ARDUINO_PIN_7           GPIO_PIN(PA, 21) /**< D7 (BLE_IRQ) */
#define ARDUINO_PIN_8           GPIO_PIN(PA, 6)  /**< D8 (BLE_CS) */
#else
#define ARDUINO_PIN_2           GPIO_UNDEF       /**< D2 */
#define ARDUINO_PIN_3           GPIO_UNDEF       /**< D3 */
#define ARDUINO_PIN_4           GPIO_UNDEF       /**< D4 */
#define ARDUINO_PIN_7           GPIO_UNDEF       /**< D7 */
#define ARDUINO_PIN_8           GPIO_UNDEF       /**< D8 */
#endif
```